src/modules/api/user/user.controller.ts

```diff
@@ -1,7 +1,35 @@
-import { Controller } from '@nestjs/common';
+import { Controller, Get, Param } from '@nestjs/common';
 import { UserService } from './user.service';
+import { IResult } from 'src/common/interfaces/response';
 
-@Controller('users/:id')
+@Controller('users')
 export class UserController {
   constructor(private userService: UserService) {}
+
+  @Get(':id')
+  async getUserById(@Param('id') id) {
+    try {
+      const user = await this.userService.getUserById(id);
+      if (user) {
+        const resp: IResult<unknown> = {
+          status: 'OK',
+          result: user,
+        };
+        return resp;
+      }
+      const errResp: IResult<unknown> = {
+        status: 'NOT OK',
```

Comment:
Write a UserDTO and use it in place of `unknown`

Reply:
Isnt it better to define a SuccessDTO and ErrorDTO and use that in place of 'unknown' ? Those could be used by all the controllers. Here instead of passing down 'user' object, we can pass down a UserDTO.